src/osvr/Connection/DeviceToken.cpp

```diff
@@ -116,6 +116,8 @@ bool OSVR_DeviceTokenObject::releaseObject(void *obj) {
 
 void OSVR_DeviceTokenObject::setDeviceDescriptor(
     std::string const &jsonString) {
+    auto guard = getSendGuard();
+    guard->lock();
```

Comment:
Do we need to call `unlock()` or does a destructor take care of that? (I gave up tracing through the code after 10 minutes.)